src/Symfony/Component/Yaml/Dumper.php

```diff
@@ -122,6 +123,15 @@ public function dump(mixed $input, int $inline = 0, int $indent = 0, int $flags
 
                 $willBeInlined = $inline - 1 <= 0 || !\is_array($value) && $dumpObjectAsInlineMap || !$value;
 
+                if (!$willBeInlined && $compactNestedMapping && \is_array($value)) {
```

Comment:
Can’t we modify the `sprintf()` below instead?

Reply:
I thought it would be less readable. Moved compact mode logic code into `sprintf`. [61c6055](https://github.com/symfony/symfony/pull/59315/commits/61c6055a40d0d5131c42e196d73977a8c3279332)